video/graphics.h

```diff
@@ -261,31 +315,95 @@ void plotPoint() {
 
 // Triangle plot
 //
-void plotTriangle(uint8_t mode) {
+void plotTriangle() {
 	Point p[3] = {
 		p3,
 		p2,
 		p1, 
 	};
-	canvas->setBrushColor(gfg);
 	canvas->fillPath(p, 3);
-	canvas->setBrushColor(tbg);
+}
+
+// Rectangle plot
+//
+void plotRectangle() {
+	canvas->fillRectangle(p2.X, p2.Y, p1.X, p1.Y);
+}
+
+// Parallelogram plot
+//
+void plotParallelogram() {
+	Point p[4] = {
+		p3,
+		p2,
+		p1,
+		Point(p1.X + (p3.X - p2.X), p1.Y + (p3.Y - p2.Y)),
+	};
+	canvas->fillPath(p, 4);
 }
 
 // Circle plot
 //
-void plotCircle(uint8_t mode) {
-	switch (mode) {
-		case 0x00 ... 0x03: { // Circle
-			auto r = 2 * (p1.X + p1.Y);
-			canvas->drawEllipse(p2.X, p2.Y, r, r);
-		} break;
-		case 0x04 ... 0x07: { // Circle
-			auto a = p2.X - p1.X;
-			auto b = p2.Y - p1.Y;
-			auto r = 2 * sqrt(a * a + b * b);
-			canvas->drawEllipse(p2.X, p2.Y, r, r);
-		} break;
+void plotCircle(bool filled = false) {
+	auto size = 2 * sqrt(rp1.X * rp1.X + rp1.Y * rp1.Y);
+	if (filled) {
+		canvas->fillEllipse(p2.X, p2.Y, size, size);
+	} else {
+		canvas->drawEllipse(p2.X, p2.Y, size, size);
+	}
+}
```

Comment:
the previous "relative" variant of circle wasn't properly calculating the size

one thing to note here is that most Agon machines are being used on a 16:9 screen, but our screen modes are essentially use a 4:3 aspect ratio, and thus have non-square pixels.  this means circles are squished when they're drawn.

this _can_ be compensated for by multiplying the height (second `size` argument) by 1.3333333 (or dividing the width by that amount).

it feels wrong to just blindly squish all circles, as that would make them incompatible with Acorn's circle plots.  however it also feels wrong that we're not providing a way to draw circles that will appear round on a 16:9 screen.

there's three potential ways to address this.  firstly we could expose fab-gl's "squished circle" ellipse variant under a set of PLOT numbers that's not already assigned (Acorn never used the range &D8-E7 - this is my least favourite option).  secondly we could implement Acorn compatible "ellipse" commands, thus allowing stretched circles to be drawn within the existing PLOT numbers, and document examples of how to draw round circles.  finally we could extend the "logical screen scaling" VDU command to look for a second bit to indicate "16:9 circle compensation" (`VDU 23,0,&C0,3` would compensate circles within the logical coordinate system, and `VDU 23,0,&C0,2` would compensate circles for native screen coordinates)